AWSIoT/Internal/AWSIoTMQTTClient.m

```diff
@@ -547,6 +576,10 @@ - (void)reconnectToSession {
     
     self.connectionAgeInSeconds = 0;
     if (self.connectionAgeTimer != nil ) {
+        // NOTE: This does not work as intended. The reconnection attempt is always done on the reconnect thread
+        // but the timer needs to be invalidated on the streams thread to ensure memory is released.
+        // However, as currently implemented a switch to the streams thread here to clean this up is always beaten to the
+        // invalidation by other events which are are already being processed on the streams thread.
         [self.connectionAgeTimer invalidate];
```

Comment:
As noted this doesn't really work but other code avoids a memory leak. Ideally, this would have code to switch threads to clean up correctly but I left this alone for now as by the time a thread switch got control another event had already done the cleanup. I can update PR to have the thread switch if desired.

Reply:
For this change, rather than doing a thread switch, lets just remove this code, as it is handled in one of the other events.